src/core/chp.jl

```diff
@@ -515,11 +518,18 @@ function get_chp_defaults_prime_mover_size_class(;hot_water_or_steam::Union{Stri
 end
 
 function get_heuristic_chp_size_kw(prime_mover_defaults_all, avg_boiler_fuel_load_mmbtu_per_hour, 
-                                prime_mover, size_class, hot_water_or_steam, boiler_effic)
-    therm_effic = prime_mover_defaults_all[prime_mover]["thermal_efficiency_full_load"][hot_water_or_steam][size_class+1]
+                                prime_mover, size_class, hot_water_or_steam, boiler_effic, thermal_efficiency=NaN)
+    if isnan(thermal_efficiency)
+        therm_effic = prime_mover_defaults_all[prime_mover]["thermal_efficiency_full_load"][hot_water_or_steam][size_class+1]
+    else
+        therm_effic = thermal_efficiency
+    end
     if therm_effic == 0.0
+        if prime_mover == "micro_turbine" && isnothing(size_class)
+            size_class = "any"
+        end
```

Comment:
given there's an error thrown after this, do we need this code? or should a warning be given instead?

Reply:
The size_class = "any" is just meant to populate the $size_class string in the error message because for micro_turbine the user may not have specified size_class, whereas for reciprocating_engine, the only way it will throw an error here is if they input a size_class of 1-3.

Reply:
that makes sense, thank you!